src/foam/nanos/auth/PasswordPolicy.js

```diff
@@ -70,6 +82,10 @@ foam.CLASS({
           javaType: 'String'
         }
       ],
+      code: function(user, potentialPassword) {
+        let password = potentialPassword.toLowerCase();
+        return this.blackList.find( w => w.toString() === password ) === undefined ? true : false;
```

Comment:
Instead of going to the Array is is O(lgn), you scanning an array which is O(n).

Reply:
Also, this will download the whole database to the client, which will probably only do one or two checks.